src/software/guides/addin-storage.md

```diff
@@ -0,0 +1,384 @@
+
+### **Sample JSON**
+
+The following sections will refer to this example JSON:
+```json
+{
+  "date": "2016-01-01T00:00:00.000Z",
+  "items": [
+    {"name": "bottles", "price": 12},
+    {"name": "caps", "price": 20}
+  ],
+  "customer": {"name": "joesmith", "email": "joe@smith.com"}
+}
+```
+
+## The purpose for and getting an AddInData GUID
+
+An AddInData GUID must be created before the Storage API methods can be used within your Add-In. This GUID is used to register and identify which AddInData implementation is associated with a given Add-In solution within Geotab.
+
+The AddInData GUID is a mandatory parameter when calling the AddInData method.
+ This allows each Add-In&#39;s data to be isolated from the data used by other Add-Ins.
+**Consider the AddInId GUID the "serial number" for your solution.**
+
+The AddInData GUID allows multiple solutions to each have their own collection of AddInData objects in the same database without the collections mixing. To generate your own GUID, please click on the following [example]({{site.baseurl}}/software/api/runner.html#sample:generate-addin-guid).
+
+## Creating an AddInData Object
+
+An AddInData object must first be created in a database. The properties of AddInData are as follows:
+
+| **Properties** |
+| --- |
+| [Id]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Id): The standard Id for any Entity. |
+| [AddInId]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Id): Used to identify the Add-In Solution to which this AddInData belongs. Add-Ins cannot see data from other Add-Ins. Must be provided when searching for AddInData. Consider this a Serial Number for this Solution that uses the Storage API. |
+| [Groups]({{site.baseurl}}/software/api/reference/#T:Geotab.Checkmate.ObjectModel.Group): Used to define the scope of a row of Add-In data. Works the same as any other ObjectModel Entity. Refer to the MyGeotab SDK [What&#39;s New](https://geotab.github.io/sdk/resources/new/) page for more information regarding the transition of this property from being required to optional. |
+| Details (String): The JSON data. May be whole or partial depending on the action (Add vs. Set) or the filtering provided when calling Get.|
+
+As an example, you can use the [API Runner tool](https://my.geotab.com/sdk/api/apiRunner.html) to create an AddInData object using the following operation:
+
+```javascript
+api.call("Add",
+{
+  "typeName": "AddInData",
+  "entity": {
+    "addInId": "a2C4ABQuLFkepPVf6-4OKAQ",
+    "groups": [
+      {
+        "id": "GroupCompanyId"
+      }
+    ],
+    "details": "{\"date\":\"2016-01-01T00:00:00.000Z\",\"items\":[{\"name\":\"bottles\",\"price\":12},{\"name\":\"caps\",\"price\":20}],\"customer\":{\"name\":\"joesmith\",\"email\":\"joe@smith.com\"}}"
```

Comment:
I think `details` can be plain json not json string like `data` property was

Reply:
Fixed